### PR TITLE
Route set up

### PR DESCRIPTION
**Pull request by akinsanmi60**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by pallabez** [comment]
Thanks for the PR. Can you please mention the description of this PR & if it was tested in the browser?

File: ui/src/App.js [hunk 1]
```diff
@@ -1,8 +1,17 @@
 import Header from "./components/Header/Header";
```
Comment (Neha) [inline ui/src/App.js:1]: The `header` is not getting used here. This should be removed from here.

File: ui/src/App.js [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
+import { Routes, Route } from "react-router-dom";
+
 
 function App() {
   return (
-    <div>Hey</div>
+    <Routes>
+      <Route path="/" />
```
Comment (Neha) [inline ui/src/App.js:8]: What would come at `/`? As of now, I can only see blank page

File: ui/src/App.js [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+      <Route path="/login" element={<Login />} />
```
Comment (Neha) [inline ui/src/App.js:9]: These components are not imported hence it is not working. only blank pages are coming.

image

File: ui/src/App.js [hunk 4]
```diff
@@ -1,8 +1,17 @@
...
+      <Route path="/flag?mode=edit" element={<FlagEdit />} />
```
Comment (Neha) [inline ui/src/App.js:10]: Same here, the components have not imported.

File: ui/src/Components/FlagMode/FlagCreate.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
```
Comment (Neha) [inline ui/src/Components/FlagMode/FlagCreate.js:1]: @pallabez are we going to have separate components for edit and create?

File: ui/src/App.js [hunk 6]
```diff
@@ -1,8 +1,17 @@
 import Header from "./components/Header/Header";
+import { Routes, Route } from "react-router-dom";
+
 
 function App() {
   return (
-    <div>Hey</div>
+    <Routes>
```
Comment (Neha) [inline ui/src/App.js:7]: This should be moved to a separate folder name `Routes`. 

@pallabez  What would be the homepage? As we are not focusing on the Login screen as of now.

File: ui/src/App.js [hunk 7]
```diff
@@ -1,8 +1,17 @@
...
+      <Route path="/" />
+      <Route path="/login" element={<Login />} />
+      <Route path="/flag?mode=edit" element={<FlagEdit />} />
+      <Route path="/flag?mode=create" element={<FlagCreate />} />
```
Comment (Neha) [inline ui/src/App.js:11]: What would be the advantage here of the `mode=create`

File: ui/src/Components/Login/Login.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import React from "react";
+
+export const Login = () => {
```
Comment (Neha) [inline ui/src/Components/Login/Login.js:3]: We are not using this pattern for component creation. Please check the `button` or `header` to follow the same pattern.

File: ui/src/Components/FlagMode/FlagEdit.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const FlagEdit = () => {
```
Comment (Neha) [inline ui/src/Components/FlagMode/FlagEdit.js:3]: We are not using this pattern for component creation. Please check the button or header to follow the same pattern.

File: ui/src/Components/FlagMode/FlagCreate.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+
+const FlagCreate = () => {
```
Comment (Neha) [inline ui/src/Components/FlagMode/FlagCreate.js:3]: We are not using this pattern for component creation. Please check the button or header to follow the same pattern.

**Review by Neha — CHANGES_REQUESTED** [review]
Please add the description and test-cases too for the PR

File: ui/src/App.js (hunk 6 above)
Comment (pallabez) [reply ui/src/App.js:7]: Homepage would be dashboard. No?

Comment (Neha) [reply ui/src/App.js:7]: Well, as we decided the login is not the priority right now then yes `dashboard`. We need to also highlight in UI till `login` is not their `dashboard` view is of the `super user` or `non-super user`.

File: ui/src/App.js (hunk 7 above)
Comment (pallabez) [reply ui/src/App.js:11]: The query would help us use same feature flag forum component to both edit & create as discussed before.

File: ui/src/Components/FlagMode/FlagCreate.js (hunk 5 above)
Comment (pallabez) [reply ui/src/Components/FlagMode/FlagCreate.js:1]: Nope. We were planning on reusing same component.

Comment (Neha) [reply ui/src/Components/FlagMode/FlagCreate.js:1]: Please remove the component and route.

**Comment by Neha** [comment]
There are new requirements for the Route. 

We would be having:

Left Nav:

1. Users
2. Flags
3. Segments
4. Environment switching
5. Account settings

and then within the pages, we will have:

1. Create a flag
2. Delete a flag
3. Edit a flag

**Comment by Neha** [comment]
Closing this PR as a new PR #64 is raised

**Pull request closed without merging**